zebrad/src/components/mempool/crawler.rs

```diff
@@ -106,6 +119,36 @@ where
             transaction_ids.len()
         );
 
-        // TODO: Send transaction IDs to the download and verify stream (#2650)
+        if !transaction_ids.is_empty() {
+            if let Err(error) = self.queue_transactions(transaction_ids).await {
+                error!(
+                    "Failed to forward crawled transactions to mempool transaction downloader: {}",
+                    error
+                );
+            }
+        }
+    }
+
+    /// Forward the crawled transactions IDs to the mempool transaction downloader.
+    async fn queue_transactions(&mut self, transaction_ids: Vec<Gossip>) -> Result<(), BoxError> {
+        let response = self
+            .mempool
+            .ready_and()
+            .await?
+            .call(mempool::Request::Queue(transaction_ids))
+            .await?;
```

Comment:
This error handling is tricky to get right, because `poll_ready` and `call` don't return any errors yet.
Here are the meanings of those errors, and how I think we should respond:

`poll_ready`/`ready_and`:
> the service is no longer able to service requests and the caller should discard the service instance.
https://docs.rs/tower-service/0.3.1/tower_service/trait.Service.html#tymethod.poll_ready

So in this case, I think the crawler should exit as well. (This is how we handle shutdowns elsewhere in Zebra.)

`call`:
https://github.com/ZcashFoundation/zebra/blob/eb99081a9a6e594635db28676477ed5d994abbe5/zebrad/src/components/mempool.rs#L291-L294

So in this case, I think the crawler should do nothing, because there is nothing the user can do about peers behaving badly. (We could log a message at debug or trace level, but not at the default `info` level, because a bad peer could spam the logs.)

What do you think?

Reply:
Sounds good. I added a commit to stop the crawler if `poll_ready` returns an error: [f668cde](https://github.com/ZcashFoundation/zebra/pull/2801/commits/f668cdeaa64976e0d4484cb079e7015397148b28).